### DISCUSSION TITLE
New to Agit

### DISCUSSION THREAD

**Discussion by mrbracchi** [0xsuk/agitcms#24] [Q&A] [answered]
Hello!

I have always done Hugo using VScode but a CMS is very intriguing to me. I installed it and took a look around and am wondering a few things.

I thought there would be options to fire off commands for pages based on a theme or widgets based on short code. For example the Academia theme from Wowchemy uses `hugo new  --kind post post/my-article-name`. I kind of expected a way for the software to see that as an option and present something in the GUI that says, "Create new post". Is that now what this is? Is Agit basically just way to preview the markdown?

**Comment by 0xsuk** [comment, marked as answer]
Thanks for feedback!

>  I thought there would be options to fire off commands for pages based on a theme

Create new post button of Agit CMS only fires certain hugo new command, so yes, I will make this command customizable via [site configuration](https://0xsuk.github.io/agitcms/configuration/#site-configuration). Site configuration applies only to specific site in your filesystem, so theme based configuration is possible.

Other than new post creation, you can use integrated terminal. Maybe you want to write shell alias in your shell config file so that can be run quickly. Also I’m planning to provide a environment variables to integrated shell so that context based scripting is possible. 


> or widgets based on short code

Can you explain a bit more about widgets? 
What kind of commands you want to run against widgets?

**Reply by mrbracchi** [reply]
Thank you for the reply. Here's an [example](https://wowchemy.com/docs/content/writing-markdown-latex/#videos) of a widget I am talking about. I do all the dev locally in VScode currently so the project lives on my machine and can be used by agit. 

I guess I was thinking it would function like blocks. So say I was found text and then wanted to embed a video, instead of using shirt code in text, there is an "embed video" button that I click and then paste the information or whatever.

**Reply by 0xsuk** [reply]
> Create new post button of Agit CMS only fires certain hugo new command

Sorry this is wrong, Create new post button (or "create new" -> "file" button correctly) does not fire hugo new command. It just creates an empty file and writes default frontmatter value to it. 

Also hugo new command works exactly the same as the create new file button of agitcms, no matter "--kind post" flag is present or not.  (hugo new command guesses which kind of content to create: https://gohugo.io/commands/hugo_new/#synopsis)

So to summarize I'm going to develop two things
- toolbar at the top of markdown editor to trigger custom function
- provide environment variable to the integrated shell so that shell scripting based on context of Agit CMS is possible. (eg: provide AGIT_EDITING_POST environment variable to the shell so that one can fire shell scripts that works based on which post Agit CMS currently edits)

**Reply by mrbracchi** [reply]
oh wow! That's cool. I am not really a developer but let me know how I can help :)

**Comment by 0xsuk** [comment]
Nice. From before I’ve been thinking about placing a toolbar at the top of the markdown editor where you can click some buttons to manipulate markdown, like making text bold.  

I’m going to offer a custom button support in the toolbar so that you can perform any task on your markdown, including short code snippet or whatnot.
